sentry/src/main/java/io/sentry/JsonSerializer.java

```diff
@@ -149,42 +149,48 @@ public <T> void serialize(@NotNull T entity, @NotNull Writer writer) throws IOEx
     writer.flush();
   }
 
+  /**
+   * Serializes an envelope to an OutputStream
+   *
+   * @param envelope the envelope
+   * @param outputStream will not be closed automatically
+   * @throws Exception an exception
+   */
   @Override
   public void serialize(@NotNull SentryEnvelope envelope, @NotNull OutputStream outputStream)
       throws Exception {
     Objects.requireNonNull(envelope, "The SentryEnvelope object is required.");
     Objects.requireNonNull(outputStream, "The Stream object is required.");
 
-    try (final BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(outputStream);
-        final Writer writer =
-            new BufferedWriter(new OutputStreamWriter(bufferedOutputStream, UTF_8))) {
-
-      envelope
-          .getHeader()
-          .serialize(new JsonObjectWriter(writer, options.getMaxDepth()), options.getLogger());
-      writer.write("\n");
-
-      for (final SentryEnvelopeItem item : envelope.getItems()) {
-        try {
-          // When this throws we don't write anything and continue with the next item.
-          final byte[] data = item.getData();
-
-          item.getHeader()
-              .serialize(new JsonObjectWriter(writer, options.getMaxDepth()), options.getLogger());
-          writer.write("\n");
-          writer.flush();
-
-          outputStream.write(data);
-
-          writer.write("\n");
-        } catch (Exception exception) {
-          options
-              .getLogger()
-              .log(SentryLevel.ERROR, "Failed to create envelope item. Dropping it.", exception);
-        }
+    // we do not want to close these as we would also close the stream that was passed in
+    final BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(outputStream);
+    final Writer writer = new BufferedWriter(new OutputStreamWriter(bufferedOutputStream, UTF_8));
+
+    envelope
+        .getHeader()
+        .serialize(new JsonObjectWriter(writer, options.getMaxDepth()), options.getLogger());
+    writer.write("\n");
+
+    for (final SentryEnvelopeItem item : envelope.getItems()) {
+      try {
+        // When this throws we don't write anything and continue with the next item.
+        final byte[] data = item.getData();
+
+        item.getHeader()
+            .serialize(new JsonObjectWriter(writer, options.getMaxDepth()), options.getLogger());
+        writer.write("\n");
+        writer.flush();
+
+        outputStream.write(data);
+
+        writer.write("\n");
+      } catch (Exception exception) {
+        options
+            .getLogger()
+            .log(SentryLevel.ERROR, "Failed to create envelope item. Dropping it.", exception);
       }
-      writer.flush();
     }
+    writer.flush();
```

Comment:
should be likely in the `finally` block?

Reply:
Only moved the code around. There's a catch block that only logs but doesn't throw out right before though.